### DISCUSSION TITLE
How to splitting code shadcn form component in react nextjs

### DISCUSSION THREAD

**Discussion by maxgsp** [community/community#124197] [Programming Help]
### Body

Hello everyone, please help me. I have just started learning Next.js and shadcn, but I am confused about how to split the code in the form component because my code has already reached 4000 lines.

How can I split it to make it shorter by breaking it into smaller components?

source code:  [https://codefile.io/f/y75KTX89uG](https://codefile.io/f/y75KTX89uG)

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by herukhoeruman** [comment]
To split it, or for code splitting, focus first on moving the UI elements. If possible, basic state like onChange and value should be reusable.
